tests/unit/src/main/java/org/mule/tck/junit4/rule/SystemProperty.java

```diff
@@ -6,13 +6,33 @@
  */
 package org.mule.tck.junit4.rule;
 
+import java.util.concurrent.Callable;
+
 import org.junit.rules.ExternalResource;
 
 /**
  * Sets up a system property before a test and guarantees to tear it down afterward.
  */
 public class SystemProperty extends ExternalResource {
 
+  /**
+   * Utility method to execute a callable with a system property and restoring the property's original value afterwards.
+   *
+   * @return the return value of the callable, or null if an exception was found during its execution.
+   */
+  public static <V> V runWithProperty(String name, String value, Callable<V> runnable) throws Throwable {
+    V callableValue;
+    SystemProperty property = new SystemProperty(name, value);
```

Comment:
Seems weird to create a SystemProperty to only call the before and after method.